### PR TITLE
Support UIImplementation

### PR DESCRIPTION
**Pull request by louischan-oursky**
ref #179

### CODE REVIEW

File: sdk/src/main/java/com/oursky/authgear/CustomTabsUIImplementation.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class CustomTabsUIImplementation : UIImplementation {
+    override fun openAuthorizationURL(
+        context: Context,
+        options: OpenAuthorizationURLOptions,
+        listener: OnOpenAuthorizationURLListener
+    ) {
+        CoroutineScope(Dispatchers.Default).launch {
```
Comment (tung2744) [inline sdk/src/main/java/com/oursky/authgear/CustomTabsUIImplementation.kt:21]: Should we let the user specify the coroutine scope?
Usually we would like to use the scope of the view model / fragement ?

File: sdk/src/main/java/com/oursky/authgear/WebKitWebViewActivity.kt [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        ): Boolean {
+            val handle = StartActivityHandle(TAG_FILE_CHOOSER, filePathCallback!!)
+            val requestCode = this.activity.handles.push(handle)
+            val intent = fileChooserParams!!.createIntent()
+            this.activity.startActivityForResult(intent, requestCode)
+            return true
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (tung2744) [inline sdk/src/main/java/com/oursky/authgear/WebKitWebViewActivity.kt:142]: I think you might need to save the webview state in onSaveInstanceState, and restore the webview state later:

https://github.com/authgear/authgear-sdk-android/blob/2023C01-24112023/sdk/src/main/java/com/oursky/authgear/latte/fragment/LatteFragment.kt#L278C24-L278C36

File: sdk/src/main/java/com/oursky/authgear/CustomTabsUIImplementation.kt (hunk 1 above)
Comment (louischan-oursky) [reply sdk/src/main/java/com/oursky/authgear/CustomTabsUIImplementation.kt:21]: I was thinking the same as when I saw this too. But AuthgearCore is using a coroutine scope that the developer cannot modify. I think we want to change this, we want to review the use of coroutine scope of the whole SDK.

File: sdk/src/main/java/com/oursky/authgear/WebKitWebViewActivity.kt (hunk 2 above)
Comment (louischan-oursky) [reply sdk/src/main/java/com/oursky/authgear/WebKitWebViewActivity.kt:142]: Seems we need to back port this fix to all SDKs too

**Pull request merged**